### PR TITLE
Chess!

### PR DESCRIPTION
**Pull request by KathrinBailey**
## About The Pull Request

Brings chess to Space Station 13!

Adds twelve jumbo chess pieces with the Staunton style to the game as structures with no density, so they can be walked through and dragged on top of other pieces. Each chess piece costs two metal, requiring 64 metal in total to build a complete chess set. Originally, the pieces were going to be crafted with titanium or plastitanium. Instead, I decided on metal. With the pieces themselves being cheap and easy to build, people can spend effort in making good-looking chess boards.

## Why It's Good For The Game
Gives players the option to craft a jumbo chess set, using an 8x8 space (+1 tile to store pieces). The Staunton design is the most common design, popularised in England in the mid 1800s, it is the most prevalent design in Europe and America. Ever look at the """"""chess"""""" programme in the holodeck and get confused at what piece is what? Did you see April and Duke Moxley use security gear as chess pieces that one time and wonder what you're even looking at? Then heck yeah, having actual chess pieces in the game is a good solution!

One thing people like to do is complain about extended. What can you do in Extended? Refurbish something, build a mini-station, build crazy power setups, make a rage cage. Why not make the intellectual rage cage? A spectator sport for all! One of those book-reading nerds of the Wizard Federation coming over? Challenge them to a game of chess! Those nerds can't refuse!

Oh, it also allows you to play chess in Space Station 13, chess is fun, it's something different to do whilst you're on the station.

## Why not just create a table that occupies one tile to play chess in?
**The complex answer:** I want chess not to be just an arcade machine that you play in the bar, I don't want this to be a round-start thing, because then the content will get stale fast. I want this to be something you have to set-up, so its a fun spectacle and it is something to work towards during a greenshift or low threat shift. Although the pieces are kind of the same, I want people to get creative and make cool new chess boards and spectating areas for it.
**The simple answer:** I don't know how to code that and I want chess that can be spectated.

## Changelog
:cl:
add: Jumbo chess!
add: Chess pieces in the metal crafting menu, use metal and scroll to find the chess pieces.
add: Chess Singularity.
/:cl:

## Images

image
image
image

_please don't crush my dreams I just wanna play chess in space with drewfus and my other friends_

### CODE REVIEW

**Comment by KathrinBailey** [comment]
I just realised that I committed the change to floors.dmi
The change so everyone is aware:
Adds a marble tile that isn't actually in the code anywhere so it cannot be yet utilised. I might make it into a stack and potentially in the future make marble something you can mine and make it an actual resource that you can construct things with, I need to dabble more in the relevant code before I can pull that off.

**Comment by KingTerrytheTerrible** [comment]
looks like a fucking downie did the sprites and code

**Comment by gardenBanter** [comment]
inferior servers stealing pp features once again smh

**Comment by KathrinBailey** [comment]
It's a very mentally deficient take to say anyone who doesn't know how to code and hasn't sprited before is a downie. Probably just as much of a mentally deficient take to say I'm stealing PPstation features when none of this is at all related to PPstation.

Yikes, brigading over a chess PR. Grow the fuck up and fuck off perhaps?

**Comment by gardenBanter** [comment]
KingTerry is one of the best programmers in his country and I have over 100 confirmed SS13 commits, every single one of which made history books. Stay MAD. The sprites are ugly (poo poo).

**Comment by KingTerrytheTerrible** [comment]
>they joined March 26 2019

**Comment by MrJWhit** [comment]
I'm a bit worried about the metal crafting page getting too large, is there a chance this could be its own crafting page?

**Comment by gardenBanter** [comment]
horrrrrrrrrrrrible sprites

**Comment by KathrinBailey** [comment]
> I'm a bit worried about the metal crafting page getting too large, is there a chance this could be its own crafting page?

I could alternatively make it craftable using different materials with much smaller crafting lists, like titanium, since titanium is usually always in excess in the ore silo.

**Comment by MrJWhit** [comment]
I think titanium is a great idea, it has a pretty low use normally.

**Comment by YakumoChen** [comment]
I like the descriptions but it could do without the coder reference tbh. the "manufactured" part doesn't even make sense anyway since you build them yourself

File: code/game/objects/structures/chess.dm [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+/obj/structure/chess
+	anchored = FALSE
+	density = FALSE
+	icon = 'icons/obj/chess.dmi'
+	icon_state = "singularity_s1"
+	name = "Singularity"
+	desc = "You've just been pranked by the Syndicate Chess Grandmaster! Report this to CentCom."
+	max_integrity = 100
+	resistance_flags = FLAMMABLE
```
Comment (Ghommie) [inline code/game/objects/structures/chess.dm:17]: Ok, it's made of metal. Why should it be flammable, copypasta?

File: code/game/objects/structures/chess.dm [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	var/buildstacktype = /obj/item/stack/sheet/metal
+	var/buildstackamount = 2
```
Comment (Ghommie) [inline code/game/objects/structures/chess.dm:19]: Useless vars, these are not windows or chairs with subtypes made of different materials.

File: code/game/objects/structures/chess.dm [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+/obj/structure/chess/wrench_act(mob/user, obj/item/tool)
+	to_chat(user, "<span class='notice'>You take apart the chess piece.</span>")
+	var/obj/item/stack/sheet/metal/M = new (loc, 2)
```
Comment (Ghommie) [inline code/game/objects/structures/chess.dm:23]: Use drop_location() instead of loc.

File: code/game/objects/structures/chess.dm [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	M.add_fingerprint(user)
+	playsound(src.loc, 'sound/items/deconstruct.ogg', 50, 1)
```
Comment (Ghommie) [inline code/game/objects/structures/chess.dm:25]:
```suggestion
	tool.play_tool_sound(src)
```

File: code/game/objects/structures/chess.dm [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+	qdel(src)
+	return TRUE
+
+/obj/structure/chess/WhitePawn
+	name = "White Pawn"
+	desc = "A white pawn manufactured by Kat-Kea. Get accused of cheating when executing a sick En Passant."
```
Comment (Ghommie) [inline code/game/objects/structures/chess.dm:31]: Could you stick with generic descriptions without all these blurps, jokes, fun facts, memes? Some of them are downright stupid, others could be moved somewhere else.

File: code/game/objects/structures/chess.dm [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+/obj/structure/chess/wrench_act(mob/user, obj/item/tool)
+	to_chat(user, "<span class='notice'>You take apart the chess piece.</span>")
+	var/obj/item/stack/sheet/metal/M = new (loc, 2)
+	M.add_fingerprint(user)
+	playsound(src.loc, 'sound/items/deconstruct.ogg', 50, 1)
+	qdel(src)
+	return TRUE
+
+/obj/structure/chess/WhitePawn
+	name = "White Pawn"
```
Comment (Ghommie) [inline code/game/objects/structures/chess.dm:30]: Use the `\improper` text macro before all these names or lowercase them.

File: code/game/objects/structures/chess.dm [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+	name = "White Queen"
+	desc = "A white queen manufactured by Kat-Kea. Gets stalked by the white knight. Imagine being beaten up by a women, hah! Fun Fact: The original Queen was the King's advisor and could only move one square diagonally."
+	icon_state = "white_queen"
+
+/obj/structure/chess/WhiteKing
+	name = "White King"
+	desc = "A white king manufactured by Kat-Kea. Fun Fact: The earliest chess piece found on Earth was made of ivory and later whale's teeth."
+	icon_state = "white_king"
+
+/obj/structure/chess/BlackPawn
```
Comment (Ghommie) [inline code/game/objects/structures/chess.dm:59]: honestly black pieces could be subtypes with different color matrix/rgb list, or just a darker hex if you don't feel like experimenting (look at material datums on tg for examples)

File: code/game/objects/structures/chess.dm [hunk 8]
```diff
@@ -0,0 +1,87 @@
+/* I don't know how to code.
+ * This is the only way I can contribute to the codebase with this shitty addition other than mapping.
+ * I don't know how to literally code chess into SS13, not even coding it so you can only do so and so moves, but I don't know how to even make a table to open a menu with little draggable pictures.
+ * Please forgive this shit code and the actual trash that this whole thing is, because adding something you can build in-game to have a little fun every 50-100 shifts or so is fun.
+ * If you are porting this, apologies, but you should check sheet_types.dm so these can be buildable at least.
+ */
```
Comment (Ghommie) [inline code/game/objects/structures/chess.dm:6]: All of this self-loathing just discourages me from merging. We are not pretending you to code the whole game, cheers.

**Review by Ghommie — CHANGES_REQUESTED** [review]
Alas, the pieces sprites stick out like a sore thumb compared to the surroundings, bulky and borderless with a shading that doesn't fit much their elavation or "flatness". Could use a flatter shading (less variated color palette), some borders and perhaps topdown perspective with less attention to minor details (they are 32x32, don't overdo yourself).

Moving on to the code...

**Comment by Ghommie** [comment]
Fix conflicts and requested changes soon, please.

**Pull request closed without merging**